formats/json/commonMain/src/kotlinx/serialization/json/internal/Polymorphic.kt

```diff
@@ -79,3 +75,8 @@ private fun throwSerializerNotFound(type: String?, jsonTree: JsonObject): Nothin
         else "class discriminator '$type'"
     throw JsonDecodingException(-1, "Polymorphic serializer was not found for $suffix", jsonTree.toString())
 }
+
+internal fun SerialDescriptor.classDiscriminator(json: Json): String =
+    annotations.filterIsInstance<JsonClassDiscriminator>().singleOrNull()?.discriminator
```

Comment:
It will create a separate `ArrayList` instance, better to avoid that and use plain loop instead

Reply:
Won't `asSequence` solve this problem?